Library/Formula/json-fortran.rb

```diff
@@ -0,0 +1,45 @@
+class JsonFortran < Formula
+  desc "A Fortran 2008 JSON API"
+  homepage "https://github.com/jacobwilliams/json-fortran"
+  url "https://github.com/jacobwilliams/json-fortran/archive/4.1.1.tar.gz"
+  sha256 "97f258d28536035ef70e9ead5c7053e654106760a12db2cc652587ed61b76124"
+
+  head "https://github.com/jacobwilliams/json-fortran.git"
+
+  option "with-unicode-support", "Build json-fortran to support unicode text in json objects and files"
+  option "without-test", "Skip running build-time tests (not recommended)"
+  option "without-robodoc", "Do not build and install ROBODoc generated documentation for json-fortran"
+
+  depends_on "robodoc" => :recommended
+  depends_on "cmake" => :build
+  depends_on :fortran
+
+  def install
+    mkdir "build" do
+      args = ["-DUSE_GNU_INSTALL_CONVENTION:BOOL=TRUE"] # Use more GNU/Homebrew-like install layout
+      args << "-DENABLE_UNICODE:BOOL=TRUE" if build.with? "unicode-support"
+      args << "-DSKIP_DOC_GEN:BOOL=TRUE" if build.without? "robodoc"
+      system "cmake", buildpath, *(args + std_cmake_args)
```

Comment:
@DomT4 I explained above but it got lost in outdated diffs… Here is the relevant text:

Fortran module files, and object code are not compatible between compilers (or sometimes even compiler releases from the same vendor). To install json-fortran on a system that wishes to support multiple Fortran compilers, it is necessary to install multiple versions of the library, each one built with the Fortran compiler it supports. To achieve this, the json-fortran project/library defaults to installing as:

```
<DESTDIR | PREFIX>/jsonfortran-${CMAKE_Fortran_COMPILER_ID}-${VERSION}/
├── cmake
│   ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config-version.cmake
│   ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config.cmake
│   ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets-release.cmake
│   └── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets.cmake
└── lib
    ├── json_module.mod
    ├── libjsonfortran.${VERSION}.dylib
    ├── libjsonfortran.${VERSION_MAJOR}.${VERSION_MINOR}.dylib -> libjsonfortran.${VERSION}.dylib
    ├── libjsonfortran.a
    └── libjsonfortran.dylib -> libjsonfortran.${VERSION_MAJOR}.${VERSION_MINOR}.dylib
```

Usually this means that the project is installed in `/usr/local/jsonfortran-${CMAKE_Fortran_COMPILER_ID}-${VERSION}/` and as a consequence homebrew does not create any symbolic links in `/usr/local/lib` and `/usr/local/include`. Also, since the `.mod` file is binary, and compiler specific—even in its name—AND the `lib` directory is on most compilers’ module search paths, the `.mod` file is just thrown in with the static and dynamic libraries.

Note, that If the user wishes to support *only one* Fortran compiler (which is what we anticipate anyone installing via Homebrew will want) and use a more standard GNU install structure they may set the CMake cache variable USE_GNU_INSTALL_CONVENTION to TRUE. This will result in the following install structure:

```
<DESTDIR | PREFIX>/
├── include
│   └── json_module.mod
└── lib
    ├── cmake
    │   └── json-fortran-${VERSION}
    │       ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config-version.cmake
    │       ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-config.cmake
    │       ├── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets-release.cmake
    │       └── jsonfortran-${CMAKE_Fortran_COMPILER_ID}-targets.cmake
    ├── libjsonfortran.${VERSION}.dylib
    ├── libjsonfortran.${VERSION_MAJOR}.${VERSION_MINOR}.dylib
    ├── libjsonfortran.a
    └── libjsonfortran.dylib
```